File: API/Models/Domain/Trail.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace API.Models.Domain
{
    public class Trail
    {
        public string Id { get; set; } = Guid.NewGuid().ToString();
        public required string Name { get; set; }
        public required string Description { get; set; }
        public string? ImageUrl { get; set; }
        public double LengthInKm { get; set; }
        public int DurationInMinutes { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public required string DifficultyId { get; set; }
        public required string RegionId { get; set; }

        // Navigation properties
        public Difficulty Difficulty { get; set; } = null!;
        public Region Region { get; set; } = null!;



    }
}

File: client/src/types/trail.ts
export type Difficulty = {
  id: string;
  name: string;
};

export type Region = {
  id: string;
  code: string;
  name: string;
};

export type Trail = {
  id: string;
  name: string;
  description: string;
  createdAt: string; 
  difficulty: Difficulty;
  difficultyId: string;
  durationInMinutes: number;
  imageUrl: string;
  lengthInKm: number;
  region: Region;
  regionId: string;
};

